### PR TITLE
leverage minimal graph to build query builder with datsapaces

### PR DESCRIPTION
**Pull request by YannanGao-gs**
## Summary

Use PMCD returned by mappinganalysis to build minimal graph for query

clean/update:  https://github.com/finos/legend-studio/pull/2523

## How did you test this change?

- [ ] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

https://github.com/user-attachments/assets/e5faf4d7-faa1-4937-9cb8-4c42d1923604

enable switch to load full graph

https://github.com/user-attachments/assets/9bddea77-4b29-41a4-a9a9-87c58aad5531

### CODE REVIEW

File: packages/legend-application-query/src/components/Core_LegendQueryApplicationPlugin.tsx [hunk 1]
```diff
@@ -727,4 +761,223 @@ export class Core_LegendQueryApplicationPlugin extends LegendQueryApplicationPlu
...
+          if (dataSpaceTaggedValue) {
+            return buildGraph;
+          }
+        }
+        return undefined;
+      },
+    ];
+  }
+
+  getExtraQueryBuilderPropagateExecutionContextChangeHelper?(): QueryBuilderPropagateExecutionContextChangeHelper[] {
```
Comment (YannanGao-gs) [inline packages/legend-application-query/src/components/Core_LegendQueryApplicationPlugin.tsx:790]: HEADS-UP: create an extension method for `propagateExecutionContextChange` because for dataspace, it depends on editorStore (@finos/legend-application-query) and dataspaceQueryBuilderState (@finos/legend-extension-dsl-data-space). 
https://github.com/finos/legend-studio/pull/2523 put this function's implementaion at  module @finos/legend-extension-dsl-data-space, but it will introduce the circular dependency.

**Comment by YannanGao-gs** [comment]
HEADS-UP: `about dataspace` will not work due to missing connection

File: packages/legend-application-query/src/stores/data-space/DataSpaceQueryCreatorStore.ts [hunk 2]
```diff
@@ -182,6 +186,10 @@ export class DataSpaceQueryCreatorStore extends QueryEditorStore {
     super.initialize();
   }
 
+  override requiresGraphBuilding(): boolean {
```
Comment (YannanGao-gs) [inline packages/legend-application-query/src/stores/data-space/DataSpaceQueryCreatorStore.ts:189]: why is this needed?

File: packages/legend-query-builder/src/stores/workflows/ServiceQueryBuilderState.ts [hunk 3]
```diff
@@ -156,23 +156,25 @@ export class ServiceQueryBuilderState extends QueryBuilderState {
...
-      mapping,
-      this.graphManagerState.usableClasses,
-    );
-    // if there is no chosen class or the chosen one is not compatible
-    // with the mapping then pick a compatible class if possible
-    if (!this.class || !compatibleClasses.includes(this.class)) {
-      const possibleNewClass = getNullableFirstEntry(compatibleClasses);
-      if (possibleNewClass) {
-        this.changeClass(possibleNewClass);
+  override async propagateExecutionContextChange(
```
Comment (YannanGao-gs) [inline packages/legend-query-builder/src/stores/workflows/ServiceQueryBuilderState.ts:159]: polymorphism in OOD will make sure this specific implementation will be triggered

File: packages/legend-extension-dsl-data-space/src/graph-manager/protocol/pure/v1/V1_DSL_DataSpace_PureGraphManagerExtension.ts [hunk 4]
```diff
@@ -211,6 +215,70 @@ export class V1_DSL_DataSpace_PureGraphManagerExtension extends DSL_DataSpace_Pu
...
+      V1_deserializeDataSpaceAnalysisResult(
+        cacheResult,
+        this.graphManager.pluginManager.getPureProtocolProcessorPlugins(),
+      ).executionContexts.every(
+        (e) => e.mappingModelCoverageAnalysisResult.model !== undefined,
+      )
+    ) {
+      analysisResult = cacheResult;
+    } else {
+      notificationService?.notify(
```
Comment (YannanGao-gs) [inline packages/legend-extension-dsl-data-space/src/graph-manager/protocol/pure/v1/V1_DSL_DataSpace_PureGraphManagerExtension.ts:246]: HEADS-UP: I moved this notification inside the else block from where it was originally added above the if statement

File: packages/legend-extension-dsl-data-space/src/stores/query-builder/DataSpaceQueryBuilderState.ts [hunk 5]
```diff
@@ -362,7 +384,8 @@ export class DataSpaceQueryBuilderState extends QueryBuilderState {
     dataSpace: DataSpace,
     executionContext: DataSpaceExecutionContext,
     dataSpaceRepo: DataSpacesBuilderRepoistory | undefined,
-    onDataSpaceChange: (val: DataSpaceInfo) => void,
+    onDataSpaceChange: (val: DataSpaceInfo) => Promise<void>,
+    isLightGraphEnabled: boolean,
```
Comment (YannanGao-gs) [inline packages/legend-extension-dsl-data-space/src/stores/query-builder/DataSpaceQueryBuilderState.ts:388]: what is it for as it's not actually used

File: packages/legend-application-query/src/stores/QueryEditorStore.ts [hunk 6]
```diff
@@ -409,6 +411,10 @@ export abstract class QueryEditorStore {
     // do nothing
   }
 
+  requiresGraphBuilding(): boolean {
```
Comment (YannanGao-gs) [inline packages/legend-application-query/src/stores/QueryEditorStore.ts:414]: currently not used

File: packages/legend-graph/src/graph-manager/protocol/pure/v1/V1_PureGraphManager.ts [hunk 7]
```diff
@@ -819,6 +819,88 @@ export class V1_PureGraphManager extends AbstractPureGraphManager {
...
+      );
+      await this.initializeAndIndexElements(graph, buildInputs, options);
+      stopWatch.record(
+        GRAPH_MANAGER_EVENT.GRAPH_BUILDER_INDEX_ELEMENTS__SUCCESS,
+      );
+
+      // build types
+      buildState.setMessage(`Building domain models...`);
+      await this.buildTypes(graph, buildInputs, options);
+      stopWatch.record(
```
Comment (YannanGao-gs) [inline packages/legend-graph/src/graph-manager/protocol/pure/v1/V1_PureGraphManager.ts:871]: necessary elements: 
class, function, service, 
(path of runtime, connection, store)

Comment (YannanGao-gs) [reply packages/legend-graph/src/graph-manager/protocol/pure/v1/V1_PureGraphManager.ts:871]: await this.buildOtherElements(graph, buildInputs, options);
      stopWatch.record(
        GRAPH_MANAGER_EVENT.GRAPH_BUILDER_BUILD_OTHER_ELEMENTS__SUCCESS,
      );

to resolve dataspace executables

File: packages/legend-graph/src/graph-manager/protocol/pure/v1/V1_PureGraphManager.ts [hunk 8]
```diff
@@ -819,6 +819,101 @@ export class V1_PureGraphManager extends AbstractPureGraphManager {
...
+
+      buildState.setMessage(`Building services and execution environments...`);
+      await this.buildServices(graph, buildInputs, options);
+      stopWatch.record(
+        GRAPH_MANAGER_EVENT.GRAPH_BUILDER_BUILD_SERVICES__SUCCESS,
+      );
+
+      // build other elements
+      buildState.setMessage(`Building other elements...`);
+      await this.buildOtherElements(graph, buildInputs, options);
```
Comment (YannanGao-gs) [inline packages/legend-graph/src/graph-manager/protocol/pure/v1/V1_PureGraphManager.ts:883]: it might fail with dataspaces that contain diagram

File: packages/legend-extension-dsl-data-space/src/stores/query-builder/DataSpaceQueryBuilderState.ts [hunk 9]
```diff
@@ -84,11 +82,34 @@ export const resolveUsableDataSpaceClasses = (
...
-  const compatibleClasses = getMappingCompatibleClasses(
+  let compatibleClasses = getMappingCompatibleClasses(
     mapping,
     graphManagerState.usableClasses,
   );
+  const mappingModelCoverageAnalysisResult =
+    queryBuilderState?.dataSpaceAnalysisResult?.executionContextsIndex.get(
+      queryBuilderState.executionContext.name,
+    )?.mappingModelCoverageAnalysisResult;
+  if (
```
Comment (YannanGao-gs) [inline packages/legend-extension-dsl-data-space/src/stores/query-builder/DataSpaceQueryBuilderState.ts:95]: HEADS-UP: This function is not the same as the one in https://github.com/finos/legend-studio/pull/2523

File: packages/legend-extension-dsl-data-space/src/graph-manager/protocol/pure/v1/V1_DSL_DataSpace_PureGraphManagerExtension.ts [hunk 10]
```diff
@@ -346,56 +425,87 @@ export class V1_DSL_DataSpace_PureGraphManagerExtension extends DSL_DataSpace_Pu
...
+      analysisResult.executionContexts.find(
+        (value) => value.name === executionContext,
+      )?.mappingModelCoverageAnalysisResult ??
+      analysisResult.executionContexts.find(
+        (value) => value.mapping === mappingPath,
+      )?.mappingModelCoverageAnalysisResult;
+    const pmcd = mappingModelCoverageAnalysisResult?.model;
+    if (pmcd && projectInfo) {
+      graphEntities = pmcd.elements
+        .concat(mappingModels)
```
Comment (YannanGao-gs) [inline packages/legend-extension-dsl-data-space/src/graph-manager/protocol/pure/v1/V1_DSL_DataSpace_PureGraphManagerExtension.ts:438]: include empty mapping and runtime

File: packages/legend-application-query/src/components/Core_LegendQueryApplicationPlugin.tsx [hunk 11]
```diff
@@ -727,4 +761,222 @@ export class Core_LegendQueryApplicationPlugin extends LegendQueryApplicationPlu
...
+      ): ((editorStore: QueryEditorStore) => GeneratorFn<void>) | undefined => {
+        function* buildGraph(): GeneratorFn<void> {
+          // do nothing
+        }
+        if (editorStore instanceof ExistingQueryEditorStore) {
+          const query = editorStore.query;
+          const dataSpaceTaggedValue = query?.taggedValues?.find(
+            (taggedValue) =>
+              taggedValue.profile === QUERY_PROFILE_PATH &&
+              taggedValue.tag === QUERY_PROFILE_TAG_DATA_SPACE &&
```
Comment (YannanGao-gs) [inline packages/legend-application-query/src/components/Core_LegendQueryApplicationPlugin.tsx:778]: HEADS_UP: there is an issue with not correctly attaching dataspace tag when creating a query, therefore, this check might not work completely for every existing query

File: packages/legend-application-query/src/stores/data-space/DataSpaceQueryCreatorStore.ts [hunk 12]
```diff
@@ -277,37 +292,95 @@ export class DataSpaceQueryCreatorStore extends QueryEditorStore {
...
+        this.graphManagerState.graph,
+        queryableDataSpace.executionContext,
+        undefined,
+        this.getProjectInfo(),
+        this.applicationStore.notificationService,
+      );
+      const mappingPath = dataSpaceAnalysisResult.executionContextsIndex.get(
+        queryableDataSpace.executionContext,
+      )?.mapping;
+      if (mappingPath) {
```
Comment (YannanGao-gs) [inline packages/legend-application-query/src/stores/data-space/DataSpaceQueryCreatorStore.ts:340]: i think we should think of another way to check if minial graph is in place

File: packages/legend-application-query/src/components/Core_LegendQueryApplicationPlugin.tsx [hunk 13]
```diff
@@ -727,4 +766,259 @@ export class Core_LegendQueryApplicationPlugin extends LegendQueryApplicationPlu
...
+        isGraphBuildingNotRequired?: boolean,
+      ): (() => Promise<void>) | undefined => {
+        /**
+         * Propagation after changing the execution context:
+         * - The mapping will be updated to the mapping of the execution context
+         * - The runtime will be updated to the default runtime of the execution context
+         * - If no class is chosen, try to choose a compatible class
+         * - If the chosen class is compatible with the new selected execution context mapping, do nothing, otherwise, try to choose a compatible class
+         */
+        const propagateExecutionContextChange = async (): Promise<void> => {
```
Comment (YannanGao-gs) [inline packages/legend-application-query/src/components/Core_LegendQueryApplicationPlugin.tsx:811]: a thought: maybe we can bypass re-building graph if the mapping remains the same

File: packages/legend-extension-dsl-data-space/src/graph-manager/protocol/pure/v1/V1_DSL_DataSpace_PureGraphManagerExtension.ts [hunk 14]
```diff
@@ -349,68 +519,149 @@ export class V1_DSL_DataSpace_PureGraphManagerExtension extends DSL_DataSpace_Pu
...
+        .concat(runtimeModels)
+        .concat(dataspaceEntity)
+        // NOTE: if an element could be found in the graph already it means it comes from system
+        // so we could rid of it
+        .filter((el) => !graph.getNullableElement(el.path, false))
+        .map((el) => this.graphManager.elementProtocolToEntity(el));
+      await this.graphManager.buildGraphForQuery(
```
Comment (YannanGao-gs) [inline packages/legend-extension-dsl-data-space/src/graph-manager/protocol/pure/v1/V1_DSL_DataSpace_PureGraphManagerExtension.ts:571]: call buildGraph

File: packages/legend-extension-dsl-data-space/src/graph-manager/protocol/pure/v1/V1_DSL_DataSpace_PureGraphManagerExtension.ts [hunk 15]
```diff
@@ -249,11 +337,90 @@ export class V1_DSL_DataSpace_PureGraphManagerExtension extends DSL_DataSpace_Pu
...
+          plugins,
+        ),
+        V1_ConcreteFunctionDefinition,
+      ),
+    );
+    const functionInfos = V1_buildFunctionInfoAnalysis(
+      functionProtocols,
+      graph,
+    );
+    const dependencyFunctionInfos = V1_buildFunctionInfoAnalysis(
```
Comment (YannanGao-gs) [inline packages/legend-extension-dsl-data-space/src/graph-manager/protocol/pure/v1/V1_DSL_DataSpace_PureGraphManagerExtension.ts:389]: promise.all()

File: packages/legend-extension-dsl-data-space/src/__lib__/to-delete/DSL_DataSpace_LegendQueryNavigation_to_delete.ts [hunk 16]
```diff
@@ -39,13 +39,14 @@ export enum DATA_SPACE_TEMPLATE_QUERY_CREATOR_ROUTE_PATTERN_TOKEN {
   GAV = 'gav',
   DATA_SPACE_PATH = 'dataSpacePath',
   TEMPLATE = 'template',
+  EXECUTION_CONTEXT = 'executionContext',
```
Comment (MauricioUyaguari) [inline packages/legend-extension-dsl-data-space/src/__lib__/to-delete/DSL_DataSpace_LegendQueryNavigation_to_delete.ts:42]: dont think we need this.

**Review by MauricioUyaguari — CHANGES_REQUESTED** [review]

File: packages/legend-query-builder/src/stores/QueryBuilderState.ts [hunk 17]
```diff
@@ -339,13 +340,31 @@ export abstract class QueryBuilderState implements CommandRegistrar {
...
-    queryExeContext.mapping = PackageableElementExplicitReference.create(
-      this.executionContextState.mapping,
-    );
-    queryExeContext.runtime = runtimeValue.packageableRuntime;
+    queryExeContext.mapping = this.executionContextState.mapping.path;
+    queryExeContext.runtime = runtimeValue.packageableRuntime.value.path;
     return queryExeContext;
   }
 
+  async propagateExecutionContextChange(
```
Comment (MauricioUyaguari) [inline packages/legend-query-builder/src/stores/QueryBuilderState.ts:348]: would it make more sense to make this an abstract method instead of an extension ?

File: packages/legend-application-query/src/__lib__/DSL_DataSpace_LegendQueryNavigation.ts [hunk 18]
```diff
@@ -39,6 +39,7 @@ export enum DATA_SPACE_TEMPLATE_QUERY_CREATOR_ROUTE_PATTERN_TOKEN {
   GAV = 'gav',
   DATA_SPACE_PATH = 'dataSpacePath',
   TEMPLATE = 'template',
+  EXECUTION_CONTEXT = 'executionContext',
```
Comment (MauricioUyaguari) [inline packages/legend-application-query/src/__lib__/DSL_DataSpace_LegendQueryNavigation.ts:42]: dont think this is needed

File: packages/legend-application-query/src/stores/data-space/DataSpaceTemplateQueryCreatorStore.ts [hunk 19]
```diff
@@ -90,62 +104,188 @@ export class DataSpaceTemplateQueryCreatorStore extends QueryEditorStore {
...
+              this.graphManagerState.graph.dependencyManager
+                .numberOfDependencies,
+            graph: graph_buildReport,
+          };
+          this.applicationStore.logService.info(
+            LogEvent.create(GRAPH_MANAGER_EVENT.INITIALIZE_GRAPH__SUCCESS),
+            graphBuilderReportData,
+          );
+        } else {
+          this.graphManagerState.graph =
```
Comment (MauricioUyaguari) [inline packages/legend-application-query/src/stores/data-space/DataSpaceTemplateQueryCreatorStore.ts:184]: any reason this else is needed, dont we already build the full graph if there is an error below ?

File: packages/legend-application-query/src/stores/LegendQueryApplicationPlugin.tsx [hunk 20]
```diff
@@ -63,6 +65,10 @@ export type NewQueryNavigationPath = (
   editorStore: ExistingQueryEditorStore,
 ) => string | undefined;
 
+export type QueryGraphBuilderGetter = (
```
Comment (MauricioUyaguari) [inline packages/legend-application-query/src/stores/LegendQueryApplicationPlugin.tsx:68]: where is this used ?

File: packages/legend-application-query/src/components/QueryEditor.tsx [hunk 21]
```diff
@@ -470,13 +470,21 @@ const QueryEditorExistingQueryInfoModal = observer(
                   <div className="query-preview__field">
                     <div className="query-preview__field__label">Mapping</div>
                     <div className="query-preview__field__value">
-                      {executionContext.mapping.value.name}
+                      {
+                        updateState.editorStore.graphManagerState.graph.getMapping(
+                          executionContext.mapping,
```
Comment (MauricioUyaguari) [inline packages/legend-application-query/src/components/QueryEditor.tsx:475]: if all we need is the name doesn't just `executionContext.mapping` work ?

File: packages/legend-application-query/src/components/data-space/DataSpaceInfo.tsx [hunk 22]
```diff
@@ -102,6 +102,8 @@ export const QueryEditorDataspaceInfoModal = observer(
       executionContext.name,
     )?.runtimeMetadata;
     const connection =
+      // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
```
Comment (MauricioUyaguari) [inline packages/legend-application-query/src/components/data-space/DataSpaceInfo.tsx:105]: why are we disabling eslint, can we not do this for any of this PR

**Review by MauricioUyaguari — CHANGES_REQUESTED** [review]

**Review by MauricioUyaguari — COMMENTED** [review]
we need a better strategy for the things we have started caching that just a bunch of ifs and checking for info. I think we should change the source of truth for UI to be dataspace info on some of the dataspace specific feature.   Im worried about the cost to maintain a lot of this code or even to test it is great. we are not accounting for a lot of things occurring. 

I think how we built the dataspace info will be determined by whether we get the info from the full graph or the cache dataspace analysis result. 
We may want to have a method to determine that this dataspace analysis result is expected to have all this new info (executable and template stuff), maybe adding a version on analysis result and saying if it has version x then we know it has all these fields are expected. Or maybe if the minimal graph is there then we know that it is expected to have this new info. If it has all this new info then we use the cached datasapce info to built it, if not we use the full graph to build this dataspace info. 
From the dataspace info is all we need to render these helper forms.

File: packages/legend-application-query/src/stores/LegendQueryApplicationPlugin.tsx (hunk 20 above)
Comment (YannanGao-gs) [reply packages/legend-application-query/src/stores/LegendQueryApplicationPlugin.tsx:68]: This is used in https://github.com/finos/legend-studio/pull/3488/files#diff-5dae4710ada9d402191df7256d5cc7d4502e3b40a8f1b9acf960143f7ee6adfeR690
This is for loading an empty function to aovid `buildFullGraph` for ExistingQueryEditorStore with dataspace context only

File: packages/legend-application-query/src/__lib__/DSL_DataSpace_LegendQueryNavigation.ts (hunk 18 above)
Comment (YannanGao-gs) [reply packages/legend-application-query/src/__lib__/DSL_DataSpace_LegendQueryNavigation.ts:42]: This is needed as when trying to build minimal graph, mapping and pmcd are fetched from corresponding sexecutionContext key

File: packages/legend-query-builder/src/stores/QueryBuilderState.ts (hunk 17 above)
Comment (YannanGao-gs) [reply packages/legend-query-builder/src/stores/QueryBuilderState.ts:348]: Other child classes e.g. ServiceQueryBuilderState will have their own implementation, the reason why I have to add an extension is that for DataspaceQueryBuilderState, I need to get access to QueryEditorStore and DataspaceQueryBuilderState at the same time, this will introduce an circular dependency in @finos/legend-extension-dsl-data-space

File: packages/legend-application-query/src/__lib__/DSL_DataSpace_LegendQueryNavigation.ts (hunk 18 above)
Comment (MauricioUyaguari) [reply packages/legend-application-query/src/__lib__/DSL_DataSpace_LegendQueryNavigation.ts:42]: Still don't understand. DataspaceInfo should have the template info( which we get from the template id) which has the executon context we need ? right

File: packages/legend-graph/src/graph-manager/action/query/Query.ts [hunk 23]
```diff
@@ -37,8 +33,9 @@ export class QueryParameterValue {
 export abstract class QueryExecutionContext {}
 
 export class QueryExplicitExecutionContext extends QueryExecutionContext {
-  mapping!: PackageableElementReference<Mapping>;
-  runtime!: PackageableElementReference<PackageableRuntime>;
+  // NOTE: Query can be built before the actual graph is built so we can't have the reference of metamodels here
+  mapping!: string;
```
[the remark this answers is not in the document]
Comment (YannanGao-gs) [reply packages/legend-graph/src/graph-manager/action/query/Query.ts:37]: setUpEdtiorState() has been updated to fetch query instead of LightQuery anymore
https://github.com/finos/legend-studio/pull/3488/files#diff-5dae4710ada9d402191df7256d5cc7d4502e3b40a8f1b9acf960143f7ee6adfeR1293

Therefore, it needs to be changed to use string

File: packages/legend-application-query/src/__lib__/DSL_DataSpace_LegendQueryNavigation.ts (hunk 18 above)
Comment (YannanGao-gs) [reply packages/legend-application-query/src/__lib__/DSL_DataSpace_LegendQueryNavigation.ts:42]: This is the entry point of building DataspaceAnalyticsResult and light graph is built when building DataspaceAnalyticsResult.
Based on current implementation, template info is built at the last step

File: packages/legend-application-query/src/components/Core_LegendQueryApplicationPlugin.tsx [hunk 24]
```diff
@@ -725,4 +764,274 @@ export class Core_LegendQueryApplicationPlugin extends LegendQueryApplicationPlu
...
+            ) {
+              const supportBuildMinimalGraph =
+                editorStore.applicationStore.config.options
+                  .TEMPORARY__enableMinimalGraph;
+              if (
+                editorStore.enableMinialGraphForDataSpaceLoadingPerformance &&
+                supportBuildMinimalGraph
+              ) {
+                try {
+                  const stopWatch = new StopWatch();
```
Comment (MauricioUyaguari) [inline packages/legend-application-query/src/components/Core_LegendQueryApplicationPlugin.tsx:840]: additionally below code is repeated. 
We should have a buildDataspaceMinialGraph in `V1_DSL_DataSpace_PureGraphManagerExtension` that takes in minimalEntities, dataspaceInfo, executionKeyToBuild, 
and builds the required graph. I think we already have this logic in `V1_DSL_DataSpace_PureGraphManagerExtension`

**Review by MauricioUyaguari — CHANGES_REQUESTED** [review]

File: packages/legend-extension-dsl-data-space/src/components/DSL_DataSpace_LegendApplicationPlugin.tsx [hunk 25]
```diff
@@ -204,4 +206,34 @@ export class DSL_DataSpace_LegendApplicationPlugin
       },
     ];
   }
+
+  getExtraQueryBuilderFunctionHelper(
```
Comment (MauricioUyaguari) [inline packages/legend-extension-dsl-data-space/src/components/DSL_DataSpace_LegendApplicationPlugin.tsx:210]: this should also be an abstract method on QueryBuilderState and we implement in DataSpaceQueryBuilderState

File: packages/legend-application-query/src/components/Core_LegendQueryApplicationPlugin.tsx (hunk 24 above)
Comment (YannanGao-gs) [reply packages/legend-application-query/src/components/Core_LegendQueryApplicationPlugin.tsx:805]: Most of these lines are for building entities. In terms of building graph, there is only one line `await dataSpaceQueryBuilderState.graphManagerState.graphManager.buildGraph(..)`
  

For the function in `V1_DSL_DataSpace_PureGraphManagerExtension `, 
these two look similar but one takes in V1_ DataSpaceAnalysisResult,  another takes in DataSpaceAnalysisResult. 
I find it will create more chaos if I abstract logic of building entities and then building graph to a new standalone function.

**Review by MauricioUyaguari — CHANGES_REQUESTED** [review]

File: packages/legend-extension-dsl-data-space/src/graph-manager/protocol/pure/v1/V1_DSL_DataSpace_PureGraphManagerExtension.ts [hunk 26]
```diff
@@ -355,55 +510,140 @@ export class V1_DSL_DataSpace_PureGraphManagerExtension extends DSL_DataSpace_Pu
...
+        );
+        return contextProtocol;
+      },
     );
+    dataspaceEntity.defaultExecutionContext =
+      analysisResult.defaultExecutionContext;
+    dataspaceEntity.title = analysisResult.title;
+    dataspaceEntity.description = analysisResult.description;
+
+    const resolvedMappingPath =
```
Comment (MauricioUyaguari) [inline packages/legend-extension-dsl-data-space/src/graph-manager/protocol/pure/v1/V1_DSL_DataSpace_PureGraphManagerExtension.ts:542]: for this, if no executionContext provided should be use the default to find the mapping ?

File: packages/legend-application-query/src/components/__tests__/QueryEditor.dataspace.test.tsx [hunk 27]
```diff
@@ -106,3 +106,70 @@ test(
     // await waitFor(() => getByText(dataspaceViewModal, 'this is template with inline query'));
   },
 );
+
+test(
```
Comment (MauricioUyaguari) [inline packages/legend-application-query/src/components/__tests__/QueryEditor.dataspace.test.tsx:110]: [NEXT PR] we need to increase test coverage here.
1. can we test when no light graph is returned in the result. 
2. can we maybe add to the engine roundtrip and test the analysis dataspace endpoint and do unit testing on some of the build graph/ build query builder state you have build.
3. can we ensure some of the CRUD methods on dataspace queries are still working and saving as expcted.

File: packages/legend-extension-dsl-data-space/src/graph-manager/protocol/pure/v1/V1_DSL_DataSpace_PureGraphManagerExtension.ts [hunk 28]
```diff
@@ -228,6 +228,26 @@ export class V1_DSL_DataSpace_PureGraphManagerExtension extends DSL_DataSpace_Pu
     );
   }
 
+  retrieveExecutionContextFromTemplateQueryId(
+    dataSpaceAnalysisResult: PlainObject<V1_DataSpaceAnalysisResult>,
+    templateQueryId: string,
+    plugins: PureProtocolProcessorPlugin[],
+  ): string | undefined {
+    const analysisResult = V1_deserializeDataSpaceAnalysisResult(
```
Comment (MauricioUyaguari) [inline packages/legend-extension-dsl-data-space/src/graph-manager/protocol/pure/v1/V1_DSL_DataSpace_PureGraphManagerExtension.ts:236]: [Next PR] lets add a test opening a template query with dataspace (with analytics info and without).

**Review by MauricioUyaguari — APPROVED** [review]
